### ISSUE TITLE
last-updated transfomer ignored after hydration

### ISSUE THREAD

**Issue by psi-4ward** [vuejs/vuepress#1035]
- [x] I confirm that this is a issue rather than a question.

## Bug report

last-updated transfomer ignored after hydration

#### Version
^1.0.0-alpha.23

#### Steps to reproduce

config.js
```javascript
plugins: {
    '@vuepress/last-updated': {
      transformer: tstamp => new Date(tstamp)
        .toLocaleString('de', {
          weekday: 'long',
          year: 'numeric',
          month: 'long',
          day: 'numeric',
          minute: '2-digit',
          hour: '2-digit'
        })
    },
```

if I debug the return value of the transfomer callback is a correct german local string.

The state of the `<v-XXX>` (the Component within `<Content>`) holds `$page.lastUpdated: "Mittwoch, 21. November 2018, 22:17"`
but the shows `Last Updated: 11/21/2018, 10:17:00 PM`

#### What is expected?

It should show the string from the transformer.

#### Other relevant information

- Your OS: Arch Linux
- Node.js version: 10.12
- Browser version: Chrome 70
- Is this a global or local install? local
- Which package manager did you use for the install? npm

**Comment by ulivz** [comment]
But when I copied your config to test, I got a `Invalid Date`.

I recommend you to use a professional date transformer instead of the native Date API, since it would have Inconsistent performance on different platforms.

**Comment by psi-4ward** [comment]
Read the source: https://github.com/vuejs/vuepress/blob/master/packages/%40vuepress/theme-default/components/Page.vue#L77

It alrady uses toLocaleString without the plugin and takes the locale from the config.js locales

**Comment by ulivz** [comment]
Cool, thanks for providing more information, it's truly a problem and the default date transform should be moved the the last-updated plugin. Fixed at c059faa6fd19ee63ab8a9515c1b0cf349b805697 and updated docs at 8c6b06916147c0ec1bd882b4a76ce93aec8a2a94
